geometry/rotation.hpp

```diff
@@ -53,6 +89,92 @@ class Rotation : public LinearMap<FromFrame, ToFrame> {
            Multivector<double, ToFrame, rank_y> y_from_frame,
            Multivector<double, ToFrame, rank_z> z_from_frame);
 
+  // While expressing one basis in another is unambiguous to define passive
+  // rotations, rotations expressed from an axis and an angle are more
+  // confusing, and Euler angles are a constant trap.  We therefore require a
+  // tag that specifies which frame is being defined by the given angle and
+  // axis.
+  // These constructors come in pairs (with either value of DefinesFrame), like
+  // the matrix constructors above.  For each pair, one of the constructors has
+  // an additional |typename = void|, see above.
+
+  template<typename Scalar,
+           typename F = FromFrame,
+           typename T = ToFrame,
+           typename = std::enable_if_t<!std::is_same<F, T>::value>>
+  Rotation(Angle const& angle,
```

Comment:
That would require an implicit conversion (the result has to be a `Rotation<ICRFJ2000Equator, ICRFJ2000Ecliptic>` at the end of the day, which tends to break things, and would be very confusing: you'd expect a constructor to give you the type you asked for).
Moreover, that syntax doesn't sound as good for the inverse (`Rotation<DefinesFrame<A>, B>`?).